docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,284 @@
+
+.. note::
+
+    The ``buffer_shape`` must be a multiple of the ``chunk_shape`` along each axis.
+
+.. note::
+
+    You can see what compression methods are available on your installation by examining the following...
+
+    .. code-block:: python
+
+      from neuroconv.tools.nwb_helpers import AVAILABLE_HDF5_COMPRESSION_METHODS
+
+      AVAILABLE_HDF5_COMPRESSION_METHODS
+
+    .. code-block:: bash
+
+      {'gzip': 'gzip',
+       ...
+       'Zstd': hdf5plugin._filters.Zstd}
+
+    And likewise for ``AVAILABLE_ZARR_COMPRESSION_METHODS``.
+
+We can confirm these values are saved by re-printing that particular dataset configuration...
+
+.. code-block:: python
+
+    print(dataset_configuration)
+
+.. code-block:: bash
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 24 B
+
+      buffer shape : (2,)
+      expected RAM usage : 16 B
+
+      chunk shape : (1,)
+      disk space usage per chunk : 8 B
+
+      compression method : Zstd
+      compression options : {'clevel': 3}
+
+
+Interfaces and Converters
+-------------------------
+
+The normal workflow when writing an NWB file using a ``DataInterface`` or ``NWBConverter`` is simple to configure.
+
+The following example uses the :ref:`example data <example_data>` available from the testing repo...
+
+.. code-block:: python
+
+    from datetime import datetime
+
+    from dateutil import tz
+    from neuroconv import ConverterPipe
+    from neuroconv.datainterfaces import SpikeGLXRecordingInterface, PhySortingInterface
+    from neuroconv.tools.nwb_helpers import (
+        make_or_load_nwbfile,
+        get_default_backend_configuration,
+        configure_backend,
+    )
+
+    # Instantiate interfaces and converter
+    ap_interface = SpikeGLXRecordingInterface(
+        file_path=".../spikeglx/Noise4Sam_g0/Noise4Sam_g0_imec0/Noise4Sam_g0_t0.imec0.ap.bin"
+    )
+    phy_interface = PhySortingInterface(
+        folder_path=".../phy/phy_example_0"
+    )
+
+    data_interfaces = [ap_interface, phy_interface]
+    converter = ConverterPipe(data_interfaces=data_interfaces)
+
+    # Fetch available metadata
+    metadata = converter.get_metadata()
+
+    # Create the in-memory NWBFile object and retrieve a default configuration
+    backend="hdf5"
+
+    nwbfile = converter.create_nwbfile(metadata=metadata)
+    backend_configuration = converter.get_default_backend_configuration(
+        nwbfile=nwbfile,
+        backend=backend,
+    )
+
+    # Make any modifications to the configuration in this step, for example...
+    backend_configuration["acquisition/ElectricalSeriesAP/data"].compression_method = "Blosc"
+
+    # Configure and write the NWB file
+    nwbfile_path = "./my_nwbfile_name.nwb"
+    converter.run_conversion(
+        nwbfile_path=nwbfile_path,
+        nwbfile=nwbfile,
+        backend=backend,
```

Comment:
I feel like this arg shouldn't be necessary if we supply a `backend_configuration`.

Reply:
Fair - opened https://github.com/catalystneuro/neuroconv/pull/840